pkg/system/phase2_creating.go

```diff
@@ -171,6 +176,128 @@ func (r *Reconciler) ReconcilePhaseCreatingForMainClusters() error {
 	return nil
 }
 
+func (r *Reconciler) UpgradeDbPostgres16() error {
+	var err error = nil
+
+	if r.NooBaa.Status.PostgresUpdatePhase == nbv1.UpgradePhaseFinished {
```

Comment:
I think that it is possible to have  `PostgresUpdatePhase == nbv1.UpgradePhaseFinished` from the previous upgrade (12->15). you should check for the major version here

Reply:
sure